agent/envoyextensions/builtin/otel-access-logging/otel_access_logging_test.go

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package otelaccesslogging
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/hashicorp/consul/api"
+	"github.com/hashicorp/consul/envoyextensions/extensioncommon"
+)
+
+func TestConstructor(t *testing.T) {
+	makeArguments := func(overrides map[string]interface{}) map[string]interface{} {
+		m := map[string]interface{}{
+			"ProxyType": "connect-proxy",
+			"Listener":  "inbound",
+			"Config": AccessLog{
+				CommonConfig: &CommonConfig{
+					LogName: "access.log",
+					GrpcService: &GrpcService{
+						Target: &Target{
+							Service: api.CompoundServiceName{
+								Name:      "otel-collector",
+								Namespace: "default",
+								Partition: "default",
+							},
+						},
+					},
+				},
+			},
+		}
+
+		for k, v := range overrides {
+			m[k] = v
+		}
+
+		return m
+	}
+
+	cases := map[string]struct {
+		extensionName string
+		arguments     map[string]interface{}
+		expected      otelAccessLogging
+		ok            bool
+	}{
+		"with no arguments": {
+			arguments: nil,
+			ok:        false,
+		},
+		"with an invalid name": {
+			arguments:     makeArguments(map[string]interface{}{}),
+			extensionName: "bad",
+			ok:            false,
+		},
+		"invalid proxy type": {
+			arguments: makeArguments(map[string]interface{}{"ProxyType": "terminating-gateway"}),
+			ok:        false,
+		},
+		"invalid listener": {
+			arguments: makeArguments(map[string]interface{}{"Listener": "invalid"}),
```

Comment:
I think this should be `ListenerType`.

```suggestion
			arguments: makeArguments(map[string]interface{}{"ListenerType": "invalid"}),
```